### DISCUSSION TITLE
"Catch-All" filter for email?

### DISCUSSION THREAD

**Discussion by jeremymeyers** [jonaswinkler/paperless-ng#743] [Support]
Hello! I set up an email address for my scanner to email scans to, and for p-ng to ingest, so I want it to process every single email that comes to that address.

The Email setup seems to be okay, but it is not processing incoming emails.

I've set up a filter with no criteria, which I thought would be the way to tell the app to just 'process everything', to no avail.

![Web capture_11-3-2021_16827_192 168 1 111](https://user-images.githubusercontent.com/3514095/110855206-0e2c2480-8284-11eb-99e9-c56533d7846f.jpeg)

The documentation says you can set up a "catch-all" filter, but there are no instructions on how to do so anywhere that i could find.

Please help! (and also add the solution to the documentation)

**Comment by shamoon** [comment]
I think that should work. Do your logs show anything useful? Wonder if it’s IMAP or something like that

**Reply by shamoon** [reply]
Another thought: are the files are coming as email attachments, not inline attachments (since you don’t have that enabled)?

**Comment by jonaswinkler** [comment]
Well yes, the idea is that if no filters are defined, everything should go through.

**Comment by jeremymeyers** [comment]
trying to figure out how to get it to check the mail manually. I think my scheduled task is a little wonky, plus it says i am 5 hours behind server time when that isn't true (server is a NAS in the other room).  I have it set to check once per minute.  I kind of assumed that since they are IMAP accounts, it would be keeping an IMAP connection open and would therefore be aware in real time when there is a new email coming in.

EDIT: I didn't notice that the mail.log display is not in reverse chronological order.  Looks like it is in fact checking the email, but not finding an email with an attachment.

**Reply by jonaswinkler** [reply]
There's no IMAP push support in the IMAP library I'm using. Mail support in python is generally somewhat lacking. 10 minutes is already pushing it, I wouldn't set it to anything lower than that.

Regarding server time: Default time zone is UTC; If you're anywhere else, you should configure that. Looks like you're somewhere in the US. I just realized I didn't mention this anywhere in the documentation - fixing that.

Regarding the actual issue: How do you send mails to the mailbox paperless is supposed to check? Could you try sending attachments using a different client / a different account? Background: I've encountered some cases where mails with invalid headers and/or content get ignored completely for integrity reasons, and you might have run into that.

**Reply by jeremymeyers** [reply]
I think i was able to get it working with some fiddling with the email settings on the printer.  Looks like https://github.com/mjs/imapclient is a python imap library that can send server messages, BTW

**Reply by jonaswinkler** [reply]
I've searched their documentation for attachment, and couldn't find anything. I don't want to do that manually.
